indigo_app/templates/indigo_api/task_form.html

```diff
@@ -118,14 +118,17 @@
         </div>
 
         <div class="card-footer text-right">
-          {% if task.state == 'open' or task.state == 'pending_review' %}
-            <form method="POST">
-            {% csrf_token %}
-              {% if cancel_task_permission %}
-                <button class="btn btn-danger float-left" type="submit" data-confirm="Are you sure you want to cancel this task?" formaction="{% url 'cancel_task' place=place.place_code pk=task.pk %}">Cancel task</button>
-              {% endif %}
-            </form>
-          {% endif %}
+          <form method="POST">
+          {% csrf_token %}
+            {% if task.state == 'open' or task.state == 'pending_review' or task.state == 'blocked' and cancel_task_permission %}
```

Comment:
This lands up being `if A or B or C and D` == `if A or B or (C and D)` because AND his higher precedence than OR. (AND is like multiply, OR is like add).

As far as I know, you can't use brackets in expressions in django templates. So you may need to move this into the view.